src/JitInterface/src/ThunkGenerator/Program.cs

```diff
@@ -272,6 +273,88 @@ unsafe partial class CorInfoImpl
             }
             tr.WriteLine();
 
+            foreach (FunctionDecl decl in functionData)
+            {
+                string returnType = decl.ReturnType.ManagedTypeName;
+                int marshalAs = returnType.LastIndexOf(']');
```

Comment:
Only MarshalAs attributes are expected to be used inside the square brackets. Not sure if we can handle string[], though this is not intended to be a general-purpose thunk generator. It only needs to handle types used in JIT-EE interface.